src/core/raster/qgsrasterlayer.h

```diff
@@ -121,6 +123,14 @@ class CORE_EXPORT QgsRasterLayer : public QgsMapLayer
        */
       QgsCoordinateTransformContext transformContext = QgsCoordinateTransformContext();
 
+
+      /**
+       * If TRUE, the layer extent will be read from XML (i.e. stored in the
+       * project file). If FALSE, the extent will be determined by the provider on layer load.
+       * \since QGIS 3.16
+       */
+      bool readExtentFromXml = false;
```

Comment:
Can we use flags here instead? I'm not a fan of adding setReadExtentFromXml/readExtentFromXml to the already complex layer api